### PR TITLE
feat(waitlist): extend wait list to all interviews

### PR DESCRIPTION
**Pull request by aaliakseyenka**
#### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Performance optimization
- [x] Refactoring
- [ ] Test Case
- [ ] Documentation update
- [ ] Other

#### 🔗 Related issue link

https://github.com/rolling-scopes/rsschool-app/issues/1644

#### 💡 Background and solution

extend waitlist to all types of interview

#### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Database migration is added or not needed
- [ ] Documentation is updated/provided or not needed
- [ ] Changes are tested locally

### CODE REVIEW

**Comment by aaliakseyenka** [comment]
@aaliakseyenka  plz hide link to waitlist for regular interviews until they are started and do not allow take student until this date

**Review by valerydluski — APPROVED** [review]

File: client/src/pages/course/mentor/interviews.tsx [hunk 1]
```diff
@@ -124,15 +126,23 @@ function isTechnicalScreening(item: { name: string }) {
...
-      <Button size="small" type="link" href={`/course/mentor/interview-students?course=${courseAlias}`}>
-        Available Students
-      </Button>
-    </>
-  ) : null;
+function renderFooter(interview: Interview, courseAlias: string) {
+  const { name, startDate, id } = interview;
+  const showWaitList = isTechnicalScreening({ name }) || moment(startDate).isBefore(moment());
+
+  if (showWaitList)
```
Comment (nataliagulko) [inline client/src/pages/course/mentor/interviews.tsx:133]: Please add `{}` after if for readability

File: client/src/pages/course/mentor/interviews.tsx [hunk 2]
```diff
@@ -124,15 +126,23 @@ function isTechnicalScreening(item: { name: string }) {
...
+    return (
+      <>
+        <div>Do you want to interview more students?</div> Please check
+        <Button
+          size="small"
+          type="link"
+          href={`/course/mentor/interview-students?course=${courseAlias}&interviewId=${id}`}
```
Comment (nataliagulko) [inline client/src/pages/course/mentor/interviews.tsx:140]: Suggestion for future refactoring: extract links into global constants file

File: client/src/api/api.ts [hunk 3]
```diff
@@ -5837,6 +5910,80 @@ export class CoursesEventsApi extends BaseAPI {
...
+            assertParamExists('getAvailableStudents', 'courseId', courseId)
+            // verify required parameter 'interviewId' is not null or undefined
+            assertParamExists('getAvailableStudents', 'interviewId', interviewId)
+            const localVarPath = `/courses/{courseId}/interviews/{interviewId}/students/available`
+                .replace(`{${"courseId"}}`, encodeURIComponent(String(courseId)))
+                .replace(`{${"interviewId"}}`, encodeURIComponent(String(interviewId)));
+            // use dummy base URL string because the URL constructor only accepts absolute URLs.
+            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
+            let baseOptions;
+            if (configuration) {
```
[the remark this answers is not in the document]
Comment (aaliakseyenka) [reply client/src/api/api.ts:5931]: this is auto-gen file. not worth review

File: client/src/modules/Interviews/pages/InterviewWaitingList/index.tsx [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        pagination={{ pageSize: 100 }}
+        size="small"
+        dataSource={availableStudents}
+        rowKey="id"
+        columns={[
+          {
+            title: 'Github',
+            dataIndex: 'githubId',
+            sorter: stringSorter('githubId'),
+            width: 180,
```
[the remark this answers is not in the document]
Comment (nataliagulko) [reply client/src/modules/Interviews/pages/InterviewWaitingList/index.tsx:74]: Got it, okay then

**Review by nataliagulko — APPROVED** [review]

**Pull request merged**